master/pkg/model/task.go

```diff
@@ -406,6 +406,14 @@ func (t TaskLogBatch) ForEach(f func(interface{}) error) error {
 	return nil
 }
 
+// NTSCModelDef represents a row in database for non experiment's model def.
+type NTSCModelDef struct {
+	bun.BaseModel `bun:"table:ntsc_model_definition"`
+
+	TaskID          TaskID `bun:"task_id"`
+	ModelDefinition []byte `bun:"model_definition"`
+}
```

Comment:
personally, i would name this `task_context_directory` all over, store it in a table called that, with `(task_id, context_directory)` and eventually move trials there.

Reply:
I think there is something a little awkward with this approach, since experiments can have multiple tasks but they will always have the same context directory.

Also I don't know if we could ever pay that migration cost.

Is ```ntsc_context_directory``` better? I don't wanna add more confusion to the question of what a task is.

Reply:
put it on the job instead? though that is technically not correct because checkpoint gc runs different code.

Reply:
I am not sure we should be overloading jobs beyond moving this atomic thing up/down in the queue or between resource pools.

e.g. looking to the future nested tasks proposal, entire task tree should probably be the same job, but the individual tasks within a tree could be running different code.

Reply:
so you agree with `task_context_directory`?

Reply:
i think, much to your dismay, the job concept is already getting overloaded.

Reply:
I think its pretty clear ```task_id, context_directory``` will be the structure of the table

I'll call it ```task_context_directory``` for now, we can always rename it if we build on it 

and I will add a comment explaining why it doesn't have trials